### ISSUE TITLE
The type initializer for 'Microsoft.CodeAnalysis.EditorFeatures.Lightup.ISmartRenameSessionFactoryWrapper' threw an exception

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#71410]
'ExtractMethodCodeRefactoringProvider' encountered an error and has been disabled:
```
System.TypeInitializationException : The type initializer for 'Microsoft.CodeAnalysis.EditorFeatures.Lightup.ISmartRenameSessionFactoryWrapper' threw an exception. ---> System.ArgumentException : Incorrect number of parameters supplied for lambda declaration
   at System.Linq.Expressions.Expression.ValidateLambdaArgs(Type delegateType,Expression& body,ReadOnlyCollection`1 parameters)
   at System.Linq.Expressions.Expression.Lambda[TDelegate](Expression body,String name,Boolean tailCall,IEnumerable`1 parameters)
   at System.Linq.Expressions.Expression.Lambda[TDelegate](Expression body,Boolean tailCall,IEnumerable`1 parameters)
   at Microsoft.CodeAnalysis.EditorFeatures.Lightup.LightupHelpers.CreateFunctionAccessor[T,TArg,TResult](Type type,String methodName,Type argType,TResult defaultValue)
   at Microsoft.CodeAnalysis.EditorFeatures.Lightup.ISmartRenameSessionFactoryWrapper..cctor()
   --- End of inner exception stack trace ---
   at Microsoft.CodeAnalysis.EditorFeatures.Lightup.ISmartRenameSessionFactoryWrapper.FromInstance(Object instance)
   at System.Lazy`1.CreateValue()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Lazy`1.get_Value()
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.RenameFlyoutViewModel..ctor(InlineRenameSession session,TextSpan selectionSpan,Boolean registerOleComponent,IGlobalOptionService globalOptionService,IThreadingContext threadingContext,IAsynchronousOperationListenerProvider listenerProvider,Lazy`1 smartRenameSessionFactory)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameAdornmentManager.<>c__DisplayClass17_0.<GetAdornment>b__0(InlineRenameSession session)
   at System.Runtime.CompilerServices.ConditionalWeakTable`2.GetValue(TKey key,CreateValueCallback createValueCallback)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameAdornmentManager.GetAdornment()
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameAdornmentManager.UpdateAdornments()
   at System.EventHandler`1.Invoke(Object sender,TEventArgs e)
   at async Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameService.StartInlineSessionAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.AbstractRenameCommandHandler.ExecuteCommandAsync(<Unknown Parameters>)
```

**Comment by CyrusNajmabadi** [comment]
@Cosifne can you please make the lightup resilient to failure. @veler as well

**Comment by CyrusNajmabadi** [comment]
@Cosifne can you please make the lightup resilient to failure. @veler as well

**Comment by Cosifne** [comment]
The issue itself is known in 17.9 P2. Will be fixed in 17.9P3.
The workaround is to switch to the legacy UI. 
(E.g. options -> Text Editor -> C#/VB -> Advanced -> Rename -> Where should the Rename UI be shown, change it to `Show UI as a Dashboard in top right`)

**Comment by Cosifne** [comment]
> @Cosifne can you please make the lightup resilient to failure. @veler as well

I talked to @sharwell offline about this.
The reason why it is broken is that LightUpHelper is introduced first and was not well tested. And we failed to pick the fix into the shipping branch.


In future the plan would be:
When 17.9 P2 rolls to integration CI, we can directly start to reference the interface. Later if editor folks make any changes in the interface, we will be completely broken. In that case, we will know things are wrong at the beginning.
Later we can switch back to light up helper, at that moment, the developer should manually verified it works.

In this way we can make sure this won't happen again.
